src/pepr/operator/controllers/network/generate.ts

```diff
@@ -132,5 +133,8 @@ export function generateName(policy: Allow) {
       .flat(1)
       .join("-");
 
-  return `${policy.direction}-${name}`;
+  // Add the port if it exists
+  const generatedName = policy.port ? `${policy.port}-${baseName}` : baseName;
+
```

Comment:
Open to differing opinion but I think I would prefer to not add the port if description is used - we added description partially to provide a way for short/simple netpol names instead of the generated ones. Maybe we can add port into the `Object.values` flattened array above instead?

Reply:
Thats where the problem is though. Since the creation of a VirtualService creates netpols and creates a description, it will just use that description and then run into collision when using the same selector for different ports. I guess I could add the port upstream to the description that the VirtualService logic is using?